tools/topology/topology1/sof-icl-rt711-rt1308-rt715-hdmi.m4

```diff
@@ -31,6 +31,18 @@ ifdef(`AMP_2_LINK',`',
 ifdef(`MIC_LINK',`',
 `define(MIC_LINK, `3')')
 
+# UAJ ID: 0, 1
+# AMP ID: 2, 3 (if EXT_AMP_REF defined)
+# DMIC ID: 4
+# HDMI ID calculated based on the configuraiton
+ifdef(`NO_LOCAL_MIC',
+  `ifdef(`NOAMP',
+       `ifdef(`NOJACK', `define(HDMI_BE_ID_BASE, `0')', `define(HDMI_BE_ID_BASE, `2')')',
+       `ifdef(`EXT_AMP_REF', `define(HDMI_BE_ID_BASE, `4')', `define(HDMI_BE_ID_BASE, `3')')'
```

Comment:
that one I don't understand. Didn't we say that we would always reverse 2 links for the amplifiers, regardless of whether the reference is present or not?

Reply:
@plbossart If we always reverse 2 links for the amplifiers, we need the patch https://github.com/thesofproject/linux/pull/3491 in kernel . Otherwise, the driver won't reverse 2 BE link ID when it is 1308. But https://github.com/thesofproject/linux/pull/3491 is not a good idea because it will impact on the existing platforms. So the driver will reserve 1 BE id for rt1308 and 2 BE ids for rt1316. This requires tplg set the HDMI BE ID based on whether there is EXT_AMP_REF or not.